mlir/lib/Dialect/LLVMIR/Transforms/AddComdats.cpp

```diff
@@ -45,12 +46,13 @@ struct AddComdatsPass : public LLVM::impl::LLVMAddComdatsBase<AddComdatsPass> {
   void runOnOperation() override {
     OpBuilder builder{&getContext()};
     ModuleOp mod = getOperation();
-    mod.walk([&](LLVM::LLVMFuncOp op) {
+    SymbolTable symbolTable{mod};
```

Comment:
One slight more tweak, can we make this lazy? Maybe:
```
std::unique_ptr<SymbolTable> symbolTable;
auto getSymTab = [&] () -> SymbolTable& {
  if (!symbolTable)
    symbolTable = std::make_unique<SymbolTable>(mod);
  return *symbolTable;
};
```

And then you can call `addComdat(op, builder, getSymTab(), mod);`